### DISCUSSION TITLE
Job name is not interpolated if the job is skipped

### DISCUSSION THREAD

**Discussion by martin-g** [community/community#101546] [Actions]
### Select Topic Area

Bug

### Body

Hello,

I have a Github Actions workflow with two jobs - https://github.com/martin-g/bioconda-containers/pull/8/files#diff-1db3b2f087c7978b9225db8364bd9324d5cd276f52986043022d73de55d1b05c.

Both jobs use strategy matrix and their names are constructed from the matrix values, e.g. `name: Build image - ${{ matrix.image }}`.

The first job always runs and its name is properly interpolated, e.g. `Build image - bioconda/bioconda-utils-build-env-cos7-aarch64`.

The second job depends on the first one (via `needs`) and uses `if` to run only when the push is to the `main` branch - https://github.com/martin-g/bioconda-containers/pull/8/files#diff-1db3b2f087c7978b9225db8364bd9324d5cd276f52986043022d73de55d1b05cR41
When the job is skipped, i.e. the push is to a branch different than `main` then its name is not interpolated, i.e. it is printed as `quay.io/bioconda/${{ matrix.DOCKER_MANIFEST }}:${{ matrix.DOCKER_TAG }}`

![github-actions-skipped-not-interpolated](https://github.com/community/community/assets/232002/616b87d5-6d87-4b82-a172-e5310a6bb406)

If I comment out the `if` or if the push is to `main` then the job name is interpolated - https://github.com/martin-g/bioconda-containers/actions/runs/7622235752

Is this a bug or a normal behavior ?

Thanks!

**Comment by IlyaGulya** [comment]
Same issue for us
<img width="742" alt="image" src="https://github.com/user-attachments/assets/ce89161a-fe42-481e-a297-0902a01facef">
